ytypes/validate.go

```diff
@@ -33,11 +52,19 @@ func Validate(schema *yang.Entry, value interface{}) util.Errors {
 		return util.NewErrs(fmt.Errorf("nil schema for type %T, value %v", value, value))
 	}
 
+	var leafrefOpt *LeafrefOptions
+	for _, o := range opts {
```

Comment:
Recommend making this a utility function (say in util/common.go). One possible way would be to add a constructor like NewLeafrefOptions(lro ...ygot.ValidationOptions). Then you could do ValidateLeafRefData(schema, value, NewLeafrefOptions(lro)). Just an idea, feel free to take it or leave it ;-)

Reply:
I can see the value of a utility function here. Just not sure I understand the semantics. You're saying here that this would just return the first `LeafrefOptions` from the `ValidationOptions`?

Reply:
Yes, the first or maybe all?

Reply:
The only issue with this is that as we add new different types of options then we will end up walking the list of options multiple times, since we'll have NewLeafrefOpts, NewFoobarOpts, etc. I'll add a TODO here.